core/utils/poly-look/src/visualisations/charts/flow-charts/sankeyDiagram/sankeyDiagram.js

```diff
@@ -0,0 +1,216 @@
+import { Chart } from "../../chart";
+
+import * as d3 from "d3";
+import * as d3Sankey from "d3-sankey";
+
+//default values
+const linkSource = ({ source }) => source,
+  linkTarget = ({ target }) => target,
+  linkValue = ({ value }) => value,
+  intern = (value) =>
+    value !== null && typeof value === "object" ? value.valueOf() : value,
+  nodeWidth = 5,
+  nodePadding = 5,
+  linkMixBlendMode = "multiply",
+  linkPath = d3Sankey.sankeyLinkHorizontal(),
+  defaultNodeTitle = (d) => `${d.id}: ${d.value}`,
+  defaultMargin = {
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  },
+  defaultLinkOpacity = 0.5,
+  defaultNodeLabelTextOpacity = 1,
+  defaultTextColor = "#0F1938",
+  defaultLinkColor = "blue",
+  defaultNodeColor = "black",
+  nodeLabelPadding = 10,
+  nodeLabelBoxPaddingX = 4,
+  nodeLabelBoxPaddingY = 1,
+  nodeLabelBoxBorderRadius = 2,
+  defaultNodeLabelBoxColor = "white",
+  defaultNodeLabelBoxOpacity = 0.7;
+
+/**
+ * Visualizes data as a flow from n to m (to x to y ..) entitites
+ *
+ * @class
+ * @param {Object[]} links - The data to be visualized as a sankey diagram
+ * @param {string} links[].source - Source of the link
+ * @param {string} links[].target - Target of the link
+ * @param {number} links[].value - Value of the link
+ * @param {number = 400} [width] - The width of the svg
+ * @param {number = 300} [height] - The height of the svg
+ * @param {Object} [options] - Options
+ * @param {boolean = true} [options.labels] - Showing labels of the nodes in the diagram
+ * @param {string} [options.align] - Alignment of the diagram (defaults to justify)
+ * @param {Object} [margin] - Margins of the chart
+ * @param {number = 0} [margin.top/right/bottom/left] - The respective margins
+ * @param {Object} [color] - All color attributes
+ * @param {string/callback = "black"} [color.node] - Color of the nodes
+ * @param {string/callback = "#0F1938"} [color.text] - Color of the label text of the nodes
+ * @param {number/callback = 1} [color.textOpacity] - Opacity of the label text of the nodes
+ * @param {string/callback = "white"} [color.nodeLabelBoxColor] - Color of the box surrounding the node label
+ * @param {number/callback =0.7"} [color.nodeLabelBoxOpacity] - Opacity of the box surrounding the node label
+ * @param {string/callback = "blue"} [color.link] - Color of the link
+ * @param {number/callback = 0.5} [color.linkOpacity] - Opacity of the link
+ */
+export class SankeyDiagram extends Chart {
+  constructor({
+    selector,
+    links,
+    width = 400,
+    height = 300,
+    options,
+    margin,
+    color,
+  }) {
+    super({ selector, width, height });
+    this._links = links;
+    this._nodeAlign =
+      {
+        left: d3Sankey.sankeyLeft,
+        right: d3Sankey.sankeyRight,
+        center: d3Sankey.sankeyCenter,
+      }[options?.align] ?? d3Sankey.sankeyJustify;
+    this._labelsShowing = options.labels == false ? false : true;
```

Comment:
Um, how about `!!options.labels`? Or if you want to return `true` unless it's specifically `false` (not the current behaviour, since you used `==` which makes it equivalent to `!options.labels`, capturing all fals-y values), you could do: `options.labels !== false`.